doc/command_ref.rst

```diff
@@ -411,10 +424,52 @@ Install Command
 ---------------
 
 ``dnf [options] install <spec>...``
-    DNF makes sure that the given packages and their dependencies are installed on the system. Each ``<spec>`` can be
-    either a ``<package-spec>``, which specifies a package directly, or a path to the local rpm package, or an URL to a remote rpm package, or a ``@<group-spec>``, which specifies an (environment) group which contains it. If a given package cannot be (and is not already) installed, the exit code will be non-zero.
+    DNF makes sure that the given packages and their dependencies are installed
+    on the system. Each ``<spec>`` can be either a :ref:`<package-spec>
+    <specifying_packages-label>`, which specifies a package directly (by name,
+    or by name-version, or so on), or a path to the local rpm package, or an URL
+    to a remote rpm package, or a provide which specifies a provide of packages,
+    or a \@\ :ref:`\<group-spec>\ <specifying_groups-label>`, which specifies an
+    (environment) group which contains it. If a given package or provide cannot
+    be (and is not already) installed, the exit code will be non-zero.
+
+    If package with given version is tried to install and other package version
+    is already installed, then DNF will try to replace the original package with
+    the described package in command line.
+
+    Please make sure that you understand which package will be selected in case
+    of multiple matches (see :ref:`\specifying_packages-label`). See also
+    :ref:`\configuration_files_replacement_policy-label`.
+
+Examples
+--------
+
+``dnf install tito``
+    Install package tito (tito is package name).
+
+``dnf install ~/Downloads/tito-0.6.2-1.fc22.noarch.rpm``
+    Install local rpm file tito-0.6.2-1.fc22.noarch.rpm in ~/Downloads/
+    directory.
 
-    Please make sure that you understand which package will be selected in case of multiple matches (see :ref:`\specifying_packages-label`). See also :ref:`\configuration_files_replacement_policy-label`.
+``dnf install tito-0.5.6-1.fc22``
+    Install package with specific version. If the package is already installed it
+    will automatically try to downgrade or upgrade to specific version.
+
+``dnf install --best tito``
```

Comment:
add global options before command as we won;t support this syntax in the future. i.e. `dnf --best install tito`